examples/EncodeDecode/main.roc

```diff
@@ -0,0 +1,111 @@
+app "example"
+    packages {
+        pf: "https://github.com/roc-lang/basic-cli/releases/download/0.8.1/x8URkvfyi9I0QhmVG98roKBUs_AZRkLFwFJVJ3942YA.tar.br",
+        json: "https://github.com/lukewilliamboswell/roc-json/releases/download/0.6.3/_2Dh4Eju2v_tFtZeMq8aZ9qw2outG04NbkmKpFhXS_4.tar.br",
+    }
+    imports [pf.Stdout.{ line }, json.Core.{ json }]
+    provides [main] to pf
+
+### start snippet impl
+
+ItemKind := [
+    Text,
+    Method,
+    Function,
+    Constructor,
+    Field,
+    Variable,
+    Class,
+    Interface,
+    Module,
+]
+    implements [
+        Decoding { decoder: decodeItems },
+        Encoding { toEncoder: encodeItems },
+        Inspect,
+        Eq,
+    ]
+
+decodeItems : Decoder ItemKind fmt
+decodeItems = Decode.custom \bytes, _ ->
+
+    # helper to convert our tag to a DecodeResult
+    ok : _, List U8 -> DecodeResult ItemKind
+    ok = \tag, rest -> { result: Ok (@ItemKind tag), rest }
+
+    when bytes is
+        ['1', .. as rest] -> ok Text rest
+        ['2', .. as rest] -> ok Method rest
+        ['3', .. as rest] -> ok Function rest
+        ['4', .. as rest] -> ok Constructor rest
+        ['5', .. as rest] -> ok Field rest
+        ['6', .. as rest] -> ok Variable rest
+        ['7', .. as rest] -> ok Class rest
+        ['8', .. as rest] -> ok Interface rest
+        ['9', .. as rest] -> ok Module rest
+        _ -> { result: Err TooShort, rest: bytes }
```

Comment:
Do we have the ability to give other errors? For example if it tries to decode "a". That isn't too short. It just doesn't match at all.

Reply:
No we don't, `TooShort` is the only error available for now in the builtins.